### PR TITLE
allow mounts on elevators

### PR DESCRIPTION
**Pull request by Gamemechanicwow**
## 🍰 Pullrequest

Allows mounting on elevators and still keeps the restriction of no mounting on boats and zeppelins.

This affects elevators at Thunder Bluff, Freewind Post, Undercity, and The Cauldron in Searing gorge. All other elevators are restricted due to being indoors.

### Proof

- https://youtu.be/Wkbwmy-Ht6Y?t=137 - Mount on Thunder Bluff elevator - Uploaded 26. sep. 2006 - Patch 1.12.0 
- https://youtu.be/qOxnmvr5uwQ?t=117 - Mount on Freewind Post elevator - Uploaded 22. sep. 2006 - Patch 1.12.0 
- https://youtu.be/gN3Z2vVNQAE?t=1792 - Mount on Undercity undervator - Uploaded 21. mar. 2014 - Joanas 1-60 are recorded during Patch 1.10

### Issues

- (https://github.com/Wall-core/Everlook-Bugtracker/issues/432)

### How2Test

- Mounting on various types of elevators.
- Being mounted and moving onto elevators.

### Todo / Checklist

- [X] None

### CODE REVIEW

File: src/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -6972,7 +6972,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                 if (m_casterUnit->IsInWater() && (!m_casterUnit->IsPlayer() || static_cast<Player*>(m_casterUnit)->IsInHighLiquid()))
                     return SPELL_FAILED_ONLY_ABOVEWATER;
 
-                if (m_casterUnit->IsPlayer() && static_cast<Player*>(m_casterUnit)->GetTransport())
+                if (m_casterUnit->IsPlayer() && (static_cast<Player*>(m_casterUnit)->GetTransport() && static_cast<Player*>(m_casterUnit)->GetTransport()->IsMoTransport()))
```
Comment (ratkosrb) [inline src/game/Spells/Spell.cpp]: Why is it needed to cast m_casterUnit to Player?

Comment (Gamemechanicwow) [reply src/game/Spells/Spell.cpp]: Good catch. I was just looking at the old conditionals and didn't think for myself.

**Comment by FlagFlayer** [comment]
Note: Joana's 1-60 was recorded during patch 1.9.2, not 1.10

**Comment by Gamemechanicwow** [comment]
> Note: Joana's 1-60 was recorded during patch 1.9.2, not 1.10

You are right. Got it mixed up. The recording is 1.9.2 but Joana's guide is for 1.10.

**Comment by ratkosrb** [comment]
Actually I think preventing mounts on transports is straight up wrong for all transports, cause I just checked and I can mount on ships in Classic. Is there any actual evidence that it wasn't possible in original Vanilla? We should just remove this whole check.

WoWScrnShot_102622_171108

**Comment by Gamemechanicwow** [comment]
well. You could sort of mount on a very small spot. 
https://www.youtube.com/watch?v=U2Ms9zqxDeg

On boats you could mount on the box. I think that one only lasted a few patches.

**Comment by Daribon** [comment]
Classic should not be taken into consideration in this matter.
Blizz decided to allow mounts on boats in classic.

I remember personally that you were only able to mount to mount on the middle crate on 2nd floor of boats in TBC. All other places on boats had disabled mounting.

**Comment by ratkosrb** [comment]
Interesting. There should probably be a moment captured in Joana's videos where she goes on a boat or zeppelin and gets dismounted then.

**Comment by Daribon** [comment]
> Interesting. There should probably be a moment captured in Joana's videos where she goes on a boat or zeppelin and gets dismounted then.

Videos from Joana show him getting dismounted before reaching the zeppelin:
Durotar zeppelin: https://youtu.be/gN3Z2vVNQAE?t=1510
Undercity zeppelin: https://youtu.be/fEPo050WscM?t=13378

It makes me wonder if not zeppelin towers should be no mount buildings. Which would be a very weird thing since iirc its taken from wmo data.
> well. You could sort of mount on a very small spot. https://www.youtube.com/watch?v=U2Ms9zqxDeg
> 
> On boats you could mount on the box. I think that one only lasted a few patches.

Here are some images I found from vanilla showing players mounted on Zeppelin, at the exact same spot as the video you linked. 
I have not found any other image that shows players mounted on Zeppelin, at any other spot.

All of the Zeppelin screenshots below are dated 2006 in filename.
WoWScrnShot_090306_165439
WoWScrnShot_090406_170331
WoWScrnShot_081506_174532
WoWScrnShot_081406_142639
WoWScrnShot_071806_234312
WoWScrnShot_072206_233658

Player attempting to summon mount just a few yards away from the spot:
WoWScrnShot_080806_165557
In this image you can even still see the error message when he attempted to find the spot:
WoWScrnShot_090206_231137
All of the other images of Zeppelins from both 2006 and 2005 does not show any player mounted.

As for boats I've found the following images from 2005 with players mounted:
WoWScrnShot_111205_133149
WoWScrnShot_121805_002212
Image taken just a few seconds earlier in 2005, showing tons of people not mounted:
WoWScrnShot_121805_002118


Image from unknown vanilla date, showing player mounted at the known spot where you would only be able to mount:
053 - zJQUw0x

Image from 2006 showing player mounted at the known place where you could mount:
WoWScrnShot_021506_165411


TL;DR Mounts were banned on zeppelins and boats, but some small spots allowed you to mount.

**Comment by GrenderG** [comment]
I can confirm it worked like that when I played Vanilla back in the day. I remember that you couldn't mount in the zeppelin, only in a small spot at the very tip of it. I fell many times trying to do it haha.

**Comment by FlagFlayer** [comment]
Why close this? Mounts should still be allowed on elevators as per the proof we have.

**Comment by Gamemechanicwow** [comment]
> Why close this? Mounts should still be allowed on elevators as per the proof we have.

It was closed because it was fixed. And looks like @brotalnia  fixed the spots! I thought hardcoding it would be a mess. But looks good! GJ
